### DISCUSSION TITLE
Multiple timeseries for same id feature selection

### DISCUSSION THREAD

**Discussion by harvetech** [blue-yonder/tsfresh#840] [Q&A] [answered]
Dear All, 
I am new to tsfresh and currently exploring feature selection. I have a dataset of 155k time series. There is a total of 20 different Ids but against each id, there are multiple time series with different labels (Regression). 
You could assume a dataset like this. It has sensors which measure the degradation of machine over time. So for the same id (machine), I have multiple degradation cycles. So how can I use feature selection in this scenario? 

your guidance will be highly appreciated. 
Best Regards
Ali Raza

**Comment by nils-braun** [comment, marked as answer]
Hi @harvetech !

You mentioned regression in your question but I am not 100% sure I understand where the regression comes into the game.
As I have understood your use-case, you have multiple time series (e.g. one time series per day) for each machine (which has a ID). For each time series you have a ground truth label if the machine is broken or not (e.g. a bool per day).

If this is the case, I would do the following: extract features for each of time series (you could just enumerate them, so you choose an artificial Id which is not the machine ID). Then maybe add some additional machine features to each set of extracted time series features and give them to the selection.

Remember: what you put into the selection needs to have the same length as your ground truth labels.
